File: submissions/p1-rats/6.pas
procedure f(x:integer);
VAR
	t:LongInt;
	p:LongInt;
BEGIN
	if x < 2000 then
			t:= 100 - (x mod 100);
			if t = 100 then t:=0
	else if (2000 <= x) and (x < 5000) then
		begin
			t:= 500 - (x mod 500);
			if t = 500 then t:=0;
		end;	
	p := x+t;	
	if x >= 5000 then 
		begin
			p:= 500 * ROUND(trunc(x*1.1) / 500);
			t:= p-x;
		end;
	
	if((t = 0) or (t > 1500))  then
		write('PAID BY CARD')
	else if(t mod 100) <> 0 then
		write('KEEP THE CHANGE YOU FILTY ANIMAL')
	else
		begin  
			write('HERE IS ');
			write(p);
		end;
END;

File: submissions/code-freeze-is-coming/6.js
function f(cost){
        var final = 0;
    var tip = 0;
    var response = '';
    if(cost < 2000){
        final = Math.ceil(cost/100)*100;
    }else if(cost < 5000){
        final = Math.ceil(cost/500)*500;
    }else{
        final = Math.round(Math.floor(cost*1.1)/500)*500;
    }
    tip = final-cost;
    if(tip > 0 && tip <= 1500){
        if((tip % 100) !== 0){
        response = 'KEEP THE CHANGE YOU FILTY ANIMAL'
      } else {
        response = 'HERE IS ' + final; 
      }
    } else {
        response = 'PAID BY CARD'
    }
    console.log(response);
}